modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm

```diff
@@ -33,212 +51,433 @@
-			var/new_style = input(owner, "Select a hair style", "Hair Alterations")  as null|anything in GLOB.hairstyles_list
+	switch(selected_alteration)
+		if("Body Colours")
+			alter_colours(alterer)
+		if("DNA")
+			alter_dna(alterer)
+		if("Hair")
+			alter_hair(alterer)
+		if("Markings")
+			alter_markings(alterer)
+
+/**
+ * Alter colours handles the changing of mutant colours
+ * This affects skin tone primarily, though has the option to change hair, markings, and mutant body parts to match
+ */
+/datum/action/innate/alter_form/proc/alter_colours(mob/living/carbon/human/alterer)
+	var/color_choice = show_radial_menu(
+		alterer,
+		alterer,
+		list(
+			"Primary" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_red"),
+			"Secondary" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_green"),
+			"Tertiary" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_blue"),
+			"All" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_rainbow"),
+		),
+		tooltips = TRUE,
+	)
+	if(!color_choice)
+		return
+	var/color_target
+	switch(color_choice)
+		if("Primary", "All")
+			color_target = "mcolor"
+		if("Secondary")
+			color_target = "mcolor2"
+		if("Tertiary")
+			color_target = "mcolor3"
+
+	var/new_mutant_colour = input(
+		alterer,
+		"Choose your character's new [color_choice = "All" ? "" : lowertext(color_choice)] color:",
+		"Form Alteration",
+		alterer.dna.features[color_target]
+	) as color|null
+	if(!new_mutant_colour)
+		return
+
+	var/marking_reset = tgui_alert(
+		alterer,
+		"Would you like to reset your markings to match your new colors?",
+		"Reset markings",
+		list("Yes", "No"),
+	)
+	var/mutant_part_reset = tgui_alert(
+		alterer,
+		"Would you like to reset your mutant body parts(not limbs) to match your new colors?",
+		"Reset mutant parts",
+		list("Yes", "No"),
+	)
+	var/hair_reset = tgui_alert(
+		alterer,
+		"Would you like to reset your hair to match your new colors?",
+		"Reset hair",
+		list("Hair", "Facial Hair", "Both", "None"),
+	)
+
+	if(color_choice == "All")
+		alterer.dna.features["mcolor"] = sanitize_hexcolor(new_mutant_colour)
+		alterer.dna.features["mcolor2"] = sanitize_hexcolor(new_mutant_colour)
+		alterer.dna.features["mcolor3"] = sanitize_hexcolor(new_mutant_colour)
+		alterer.dna.update_uf_block(DNA_MUTANT_COLOR_BLOCK)
+		alterer.dna.update_uf_block(DNA_MUTANT_COLOR_2_BLOCK)
+		alterer.dna.update_uf_block(DNA_MUTANT_COLOR_3_BLOCK)
+	else
+		alterer.dna.features[color_target] = sanitize_hexcolor(new_mutant_colour)
+		switch(color_target)
+			if("mcolor")
+				alterer.dna.update_uf_block(DNA_MUTANT_COLOR_BLOCK)
+			if("mcolor2")
+				alterer.dna.update_uf_block(DNA_MUTANT_COLOR_2_BLOCK)
+			if("mcolor3")
+				alterer.dna.update_uf_block(DNA_MUTANT_COLOR_3_BLOCK)
+
+	if(marking_reset && marking_reset == "Yes")
```

Comment:
what is the need for checking if marking_reset is something when you check if its "Yes" directly afterwards?